torch/csrc/utils/tensor_new.cpp

```diff
@@ -130,6 +132,24 @@ static Tensor new_from_sequence(const Type & type, int device, PyObject* data) {
   return new_from_data(type, device, data);
 }
 
+static void check_is_dense(const Type& type) {
+  if (type.is_sparse()) {
+    std::ostringstream oss;
+    oss << "new(..) on a dense tensor can only be called with a dense dtype, got: ";
+    oss << torch::getDtype(type)->name;
+    throw std::runtime_error(oss.str());
```

Comment:
TypeError from `Exceptions.h` might be a little better but doesn't really matter